### ISSUE TITLE
`rationalize()`: Explicitly specify polynomial generators to allow symbols in coefficients?

### ISSUE THREAD

**Issue by filonik** [josdejong/mathjs#2599]
In SymPy, you can explicitly specify the polynomial generators, which makes it possible to distinguish them from other symbols, see:

https://docs.sympy.org/latest/modules/polys/basics.html

Is there a way to emulate this in mathjs? This would be the desired behavior (not working/pseudo code):

```
const ret = math.rationalize('a*x+b', {}, {variables: ["x"], detailed: true})
// ret.expression=a*x+b, ret.variables = ["x"], ret.coefficients=[b,a]
```

My suspicion is that this currently is not possible due to limitations in the rewriting rules, but maybe there is some clever trick to make it work.

**Comment by josdejong** [comment]
Sorry I don't fully understand your idea, can you elaborate?

**Comment by filonik** [comment]
Sure. If I understand `rationalize()` correctly, the current implementation automatically assumes that any symbol in the expression is a polynomial generator. However, it can sometimes be useful to also include symbols into the domain of coefficients. The linked SymPy documentation contains some examples (e.g. using `pi` as a symbolic constant).

Perhaps, the current implementation should be left as is, and instead it would be better to provide an alternative function for these use cases.

**Comment by josdejong** [comment]
Ah, so you would like to be able to treat specific symbols like `pi` as a _constant_ rather than a _variable_ in the polynomial? That makes sense. My initial feeling is that it could be well implemented as a new option for the existing `rationalize` function.

Anyone interested into trying this out? Help would be welcome.

**Comment by filonik** [comment]
Yes, that is another great way of putting it. I suppose (one of) the biggest questions would be how to distinguish "constant" vs "variable" symbols in the rules that `rationalize()` uses to transform the expression. Provided that, my feeling is that the existing rules should work (with the only modification being to take this additional distinction into account). However, the rules are quite complex and I don't have the experience necessary to make the changes. There is certainly a possibility that things could break in subtle ways...

I just wanted to log this "feature request", since it seems feasible in principle, and it would be rather useful to have in certain practical applications.

**Comment by gwhitney** [comment]
Since this is more of a feature idea than a specifically actionable issue as it stands, moving it to Discussions.
